src/reformat_libyuv.c

```diff
@@ -335,64 +336,270 @@ avifResult avifImageRGBToYUVLibYUV8bpc(avifImage * image, const avifRGBImage * r
+            { NULL, NULL, NULL, NULL, NULL },                                     // RGB
+            { NULL, I410ToARGBMatrix, I210ToARGBMatrix, I010ToARGBMatrix, NULL }, // RGBA
+            { NULL, NULL, NULL, NULL, NULL },                                     // ARGB
+            { NULL, NULL, NULL, NULL, NULL },                                     // BGR
+            { NULL, I410ToARGBMatrix, I210ToARGBMatrix, I010ToARGBMatrix, NULL }, // BGRA
+            { NULL, NULL, NULL, NULL, NULL },                                     // ABGR
+            { NULL, NULL, NULL, NULL, NULL },                                     // RGB_565
+        },
+        // 12bpc
+        {
+            // { NONE, YUV444, YUV422, YUV420, YUV400 }   // AVIF_RGB_FORMAT_
+            { NULL, NULL, NULL, NULL, NULL },             // RGB
+            { NULL, NULL, NULL, I012ToARGBMatrix, NULL }, // RGBA
+            { NULL, NULL, NULL, NULL, NULL },             // ARGB
+            { NULL, NULL, NULL, NULL, NULL },             // BGR
+            { NULL, NULL, NULL, I012ToARGBMatrix, NULL }, // BGRA
+            { NULL, NULL, NULL, NULL, NULL },             // ABGR
+            { NULL, NULL, NULL, NULL, NULL },             // RGB_565
+        },
+    };
+
+    memset(lcf, 0, sizeof(*lcf));
+    assert(rgb->depth == 8);
+    if (yuvDepth > 8) {
+        assert(yuvDepth == 10 || yuvDepth == 12);
+        int depthIndex = (yuvDepth == 10) ? 0 : 1;
+        if (yuvFormat != AVIF_PIXEL_FORMAT_YUV444) {
+            lcf->yuvToRgbMatrixFilterHighBitDepth = lutYuvToRgbMatrixFilterHighBitDepth[depthIndex][rgb->format][yuvFormat];
+            if (lcf->yuvToRgbMatrixFilterHighBitDepth != NULL) {
+                return AVIF_TRUE;
+            }
+        }
+        if (yuvFormat == AVIF_PIXEL_FORMAT_YUV444 || nearestNeighborFilterAllowed(rgb->chromaUpsampling)) {
+            lcf->yuvToRgbMatrixHighBitDepth = lutYuvToRgbMatrixHighBitDepth[depthIndex][rgb->format][yuvFormat];
+            if (lcf->yuvToRgbMatrixHighBitDepth != NULL) {
+                return AVIF_TRUE;
+            }
+        }
+        // Fallthrough is intentional. No high bitdepth libyuv function was found. Check if there is an 8-bit libyuv function which
+        // can used with a downshift.
+    }
+    if (yuvFormat == AVIF_PIXEL_FORMAT_YUV400) {
+        lcf->yuv400ToRgbMatrix = lutYuv400ToRgbMatrix[rgb->format];
+        return lcf->yuv400ToRgbMatrix != NULL;
+    }
+    if (yuvFormat != AVIF_PIXEL_FORMAT_YUV444) {
+        lcf->yuvToRgbMatrixFilter = lutYuvToRgbMatrixFilter[rgb->format][yuvFormat];
+        if (lcf->yuvToRgbMatrixFilter != NULL) {
+            return AVIF_TRUE;
+        }
+        if (!nearestNeighborFilterAllowed(rgb->chromaUpsampling)) {
+            return AVIF_FALSE;
+        }
+    }
+    lcf->yuvToRgbMatrix = lutYuvToRgbMatrix[rgb->format][yuvFormat];
+    return lcf->yuvToRgbMatrix != NULL;
```

Comment:
Optional: We can rewrite lines 546-556 to have a similar structure as the high-bit-depth case:

```
    if (yuvFormat != AVIF_PIXEL_FORMAT_YUV444) {
        lcf->yuvToRgbMatrixFilter = lutYuvToRgbMatrixFilter[rgb->format][yuvFormat];
        if (lcf->yuvToRgbMatrixFilter != NULL) {
            return AVIF_TRUE;
        }
    }
    if (yuvFormat == AVIF_PIXEL_FORMAT_YUV444 || nearestNeighborFilterAllowed(rgb->chromaUpsampling)) {
        lcf->yuvToRgbMatrix = lutYuvToRgbMatrix[rgb->format][yuvFormat];
        if (lcf->yuvToRgbMatrix != NULL) {
            return AVIF_TRUE;
        }
    }
    return AVIF_FALSE;
```

Reply:
Note: Although I suggested the optional rewrite, I actually prefer the orginal. simpler structure of the 8-bit case, but it is a little confusing to see the high-bit-depth case look quite different from the 8-bit case.

Reply:
yes, while there is a difference between the two cases, the different is apparent once you think about the fact that the high bitdepth case falls through and the 8-bit case does not.

so i will leave this as-is.